extensions/control-plane/provision/provision-bigquery/src/main/java/org/eclipse/edc/connector/provision/gcp/BigQueryProvisioner.java

```diff
@@ -0,0 +1,196 @@
+/*
+ *  Copyright (c) 2024 Google LLC
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Google LLC
+ *
+ */
+
+package org.eclipse.edc.connector.provision.gcp;
+
+import org.eclipse.edc.connector.transfer.spi.provision.Provisioner;
+import org.eclipse.edc.connector.transfer.spi.types.DeprovisionedResource;
+import org.eclipse.edc.connector.transfer.spi.types.ProvisionResponse;
+import org.eclipse.edc.connector.transfer.spi.types.ProvisionedResource;
+import org.eclipse.edc.connector.transfer.spi.types.ResourceDefinition;
+import org.eclipse.edc.gcp.bigquery.BigQueryService;
+import org.eclipse.edc.gcp.bigquery.BigQueryServiceImpl;
+import org.eclipse.edc.gcp.bigquery.BigQueryTarget;
+import org.eclipse.edc.gcp.common.GcpAccessToken;
+import org.eclipse.edc.gcp.common.GcpConfiguration;
+import org.eclipse.edc.gcp.common.GcpException;
+import org.eclipse.edc.gcp.common.GcpServiceAccount;
+import org.eclipse.edc.gcp.iam.IamService;
+import org.eclipse.edc.gcp.iam.IamServiceImpl;
+import org.eclipse.edc.policy.model.Policy;
+import org.eclipse.edc.spi.monitor.Monitor;
+import org.eclipse.edc.spi.response.ResponseStatus;
+import org.eclipse.edc.spi.response.StatusResult;
+import org.eclipse.edc.spi.types.TypeManager;
+
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+
+import static java.util.concurrent.CompletableFuture.completedFuture;
+
+public class BigQueryProvisioner implements Provisioner<BigQueryResourceDefinition, BigQueryProvisionedResource> {
+    private final GcpConfiguration gcpConfiguration;
+    private final Monitor monitor;
+    private final TypeManager typeManager;
+    private BigQueryService bigQueryService;
```

Comment:
the `BigQueryService` is not thread safe (see the usage of `streamWriter`), so it's not safe to set it as field as the provisioner must be thread safe, either set it as local variable or make it thread safe

Reply:
Now using ExecutorService, is it ok to keep the object as field?